src/hashtable-shim/defs.lisp

```diff
@@ -1,15 +1,17 @@
-(uiop:define-package #:coalton/hashtable-shim
+(defpackage #:coalton/hashtable-shim
```

Comment:
While we're touching this could we move this under the library?

Reply:
Posterity: see https://github.com/coalton-lang/coalton/pull/344#issuecomment-1003446893 for my initial reasoning for putting it as a separate subsystem rather than in `coalton-library`.

It could easily be either a subsystem of `coalton-library`, or a module within it, though. 🤷